src/blockchain/src/block.rs

```diff
@@ -13,22 +13,54 @@
    See the License for the specific language governing permissions and
    limitations under the License.
 */
+use std::fmt::{Debug, Display, Formatter};
+use std::hash::{Hash, Hasher};
+use std::time::{SystemTime, UNIX_EPOCH};
 
+use anyhow::Error;
 use libp2p::identity;
-use rand::Rng;
 use serde::{Deserialize, Serialize};
-use std::fmt::{Display, Formatter};
-use std::hash::Hash;
-use std::time::{SystemTime, UNIX_EPOCH};
 
-use super::crypto::hash_algorithm::HashDigest;
 use super::header::*;
-use super::signature::Signature;
 
 // TransactionType define the type of transaction, currently only create
 #[derive(Serialize, Deserialize, Debug, Clone, Hash, PartialEq, Eq, Copy)]
 pub enum TransactionType {
     Create,
+    AddAuthority,
+    RevokeAuthority,
+}
+
+// struct Signature define a general structure of signature
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq)]
+pub struct Signature {
```

Comment:
You have lots of merge conflicts 😱 this was moved... its duplicated in ur branch

https://github.com/chb0github/pyrsia/blob/34b005bede9effe148287ff0b7523ecde70c5a82/src/blockchain/src/signature.rs#L26

Reply:
Didn't we resolve all this yest?